lib/rouge/regex_lexer.rb

```diff
@@ -95,10 +127,14 @@ def appended(&defn)
       #   {RegexLexer#token}, and {RegexLexer#delegate}.  The first
       #   argument can be used to access the match groups.
       def rule(re, tok=nil, next_state=nil, &callback)
+        raise ClosedState.new(self) if @closed
+
         if tok.nil? && callback.nil?
           raise "please pass `rule` a token to yield or a callback"
         end
 
+        matches_empty = re.match?('')
```

Comment:
I think this is the reason the CI builds below 2.4 are failing. `Regexp#match?` was [added in 2.4](https://www.ruby-lang.org/en/news/2016/12/25/ruby-2-4-0-released/). Did you specifically want to avoid alternatives because of the performance improvements? Is this intended as something for Rouge 4.0?

Reply:
That'd do it. Changed to `=~`.